### ISSUE TITLE
error installing app in automation:

### ISSUE THREAD

**Issue by abelsonlive** [newslynx/opportunities#99]
I get this in `app.log` when trying to login into the application after running automation:

![screenshot 2015-08-27 21 08 14](https://cloud.githubusercontent.com/assets/1621721/9536913/db5ca12a-4cff-11e5-8e96-9f68910a6596.png)

**Comment by mhkeller** [comment]
You're on the login page and not the "where is the API URL?" Page?

**Comment by mhkeller** [comment]
It's GETing the setup page and the POSTing data to it.

**Comment by abelsonlive** [comment]
yeah i got past the "where is the api url" page. I used 'http://localhost:5001".

**Comment by mhkeller** [comment]
It will redirect to setup if api_url isn't set in config or whatever that setting is called. Which will address the redirect issue. I can push a change to address this undefined error

Im case it's using your system variables, I think trashing your config.yaml and starting over from the sample in dot_newslynx would be good to make sure

**Comment by mhkeller** [comment]
this fixed the callback not being present: https://github.com/newslynx/newslynx-app/commit/38987f42aaad512daf5323a80f9a35d5d0dc25ef

if the api_url is present in config, it will go directly to the login page, which should be the automation flow.

Test again with this latest version and close if fixed

**Comment by abelsonlive** [comment]
Getting this now:

```
#####################################
# HTTP listening on 0.0.0.0:3000... #
#####################################
UsIrizogpBk5p6bGYAyZ2kpt7m5uNMWd
GET / 302 20ms - 68b
UsIrizogpBk5p6bGYAyZ2kpt7m5uNMWd
GET /login 302 6ms - 68b
UsIrizogpBk5p6bGYAyZ2kpt7m5uNMWd
GET /setup 200 349ms
GET /stylesheets/octicon/octicons.css 200 14ms
GET /stylesheets/css/setup.css 200 65ms
UsIrizogpBk5p6bGYAyZ2kpt7m5uNMWd
POST /setup 200 112ms
/opt/newslynx/newslynx-app/lib/settings/config.js:36
      cb(err)
      ^
TypeError: undefined is not a function
    at /opt/newslynx/newslynx-app/lib/settings/config.js:36:7
    at FSReqWrap.oncomplete (fs.js:95:15)
error: Forever detected script exited with code: 1
error: Script restart attempt #1
newslynx-app-config module not present. We are not in an Electron environment.

/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:166
  throw generateError(state, message);
        ^
YAMLException: end of the stream or a document separator is expected at line 2, column 15:
      "secret_key": "fo0ewrifasdjfs-bar",
                  ^
    at generateError (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:160:10)
    at throwError (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:166:9)
    at readDocument (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:1498:5)
    at loadDocuments (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:1538:5)
    at Object.load (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:1555:19)
    at Object.<anonymous> (/opt/newslynx/newslynx-app/lib/settings/config.js:28:28)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
error: Forever detected script exited with code: 1
newslynx-app-config module not present. We are not in an Electron environment.

/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:166
  throw generateError(state, message);
        ^
YAMLException: end of the stream or a document separator is expected at line 2, column 15:
      "secret_key": "fo0ewrifasdjfs-bar",
                  ^
    at generateError (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:160:10)
    at throwError (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:166:9)
    at readDocument (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:1498:5)
    at loadDocuments (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:1538:5)
    at Object.load (/opt/newslynx/newslynx-app/node_modules/js-yaml/lib/js-yaml/loader.js:1555:19)
    at Object.<anonymous> (/opt/newslynx/newslynx-app/lib/settings/config.js:28:28)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
error: Forever detected script exited with code: 1
```

**Comment by abelsonlive** [comment]
hmm... is the app somehow modifying `~/.newslnyx/config.yaml`. it looks like this:

```
module.exports = {
  "secret_key": "fo0ewrifasdjfs-bar",
  "app_secret_key": "chiken-burrito-grande",
  "super_user": "admin",
  "super_user_email": "changme@example.com",
  "super_user_password": "admin",
  "super_user_apikey": "mj",
  "default_recipes": "/root/.newslynx/defaults/recipes.yaml",
  "default_tags": "/root/.newslynx/defaults/tags.yaml",
  "sous_chefs_dir": "~/.newslynx/sous-chefs/",
  "sqlalchemy_database_uri": "postgres://postgres:changeme@localhost:5432/newslynx",
  "redis_url": "redis://localhost:6379/",
...
```

**Comment by mhkeller** [comment]
Looks like a yaml malformed error. For now set api_url in config and put this as low priority. This endpoint mostly exists for electron and I can debug that separately.

**Comment by mhkeller** [comment]
Yes it's for electron mostly and used in conjunction with the separate newslynx app config node module.

**Comment by abelsonlive** [comment]
Oh wow.. i didn't realize my local config didn't have the api url.

**Comment by mhkeller** [comment]
best to copy over the default config to avoid legacy issues

On Thu, Aug 27, 2015 at 10:14 PM, Brian Abelson notifications@github.com
wrote:

> Closed #99 https://github.com/newslynx/opportunities/issues/99.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/newslynx/opportunities/issues/99#event-394650200.

## 

@mhkeller
mhkeller.com
